File: scripts/parse_projects.py
import re, os, glob

pages_dir = "/tmp/projects"

for filepath in sorted(glob.glob(os.path.join(pages_dir, "*.html"))):
    fname = os.path.basename(filepath)
    with open(filepath, 'r', encoding='utf-8', errors='ignore') as f:
        html = f.read()
    
    print(f"===== {fname} =====")
    
    # Title
    title_m = re.search(r'<title>(.*?)</title>', html)
    title = title_m.group(1).strip() if title_m else "N/A"
    title = re.sub(r'\s*[-|]\s*Shandong Qixing.*$', '', title).strip()
    print(f"Title: {title}")
    
    # Find all hkimg image IDs
    img_ids = set()
    for m in re.finditer(r'hkimg[^"]*/(\d+)\.(png|jpg|jpeg)', html):
        img_ids.add(m.group(1))
    for m in re.finditer(r'data-src="[^"]*hkimg[^"]*/(\d+)\.(png|jpg|jpeg)', html):
        img_ids.add(m.group(1))
    for m in re.finditer(r'src="[^"]*hkimg[^"]*/(\d+)\.(png|jpg|jpeg)', html):
        img_ids.add(m.group(1))
    
    print(f"Image IDs: {sorted(img_ids)}")
    
    # Strip all tags for text
    body = re.sub(r'<script[^>]*>.*?</script>', '', html, flags=re.DOTALL)
    body = re.sub(r'<style[^>]*>.*?</style>', '', body, flags=re.DOTALL)
    body = re.sub(r'<[^>]+>', ' ', body)
    body = re.sub(r'\s+', ' ', body).strip()
    
    # Find key info
    for key in ['Customer', 'Project', 'Products', 'Operation']:
        m = re.search(rf'{key}[:\s]*([^.!?\n]{{15,200}})', body)
        if m:
            val = m.group(1).strip()
            val = re.sub(r'^\s*(:|-|—)\s*', '', val)
            print(f"  {key}: {val[:180]}")
    
    # First meaningful paragraph
    chunks = [s.strip() for s in body.split('.') if len(s.strip()) > 40]
    if chunks:
        print(f"  Text: {chunks[0][:250]}...")
    
    print()

File: scripts/parse_projects.mjs
import { readFileSync, readdirSync } from 'fs';
import { join } from 'path';

const dir = '/tmp/projects';
const files = readdirSync(dir).filter(f => f.endsWith('.html'));

for (const fname of files) {
    const html = readFileSync(join(dir, fname), 'utf-8');
    
    console.log(`===== ${fname} =====`);
    
    // Title
    const titleM = html.match(/<title>(.*?)<\/title>/);
    let title = titleM ? titleM[1].trim() : 'N/A';
    title = title.replace(/\s*[-|]\s*Shandong Qixing.*$/, '').trim();
    console.log(`Title: ${title}`);
    
    // Find all hkimg image IDs
    const imgIds = new Set();
    const imgRegex = /hkimg[^"]*\/(\d+)\.(png|jpg|jpeg)/g;
    let m;
    while ((m = imgRegex.exec(html)) !== null) {
        imgIds.add(m[1]);
    }
    console.log(`Image IDs: [${[...imgIds].sort().join(', ')}]`);
    
    // Strip all tags for text
    let body = html.replace(/<script[^>]*>[\s\S]*?<\/script>/g, '');
    body = body.replace(/<style[^>]*>[\s\S]*?<\/style>/g, '');
    body = body.replace(/<[^>]+>/g, ' ');
    body = body.replace(/\s+/g, ' ').trim();
    
    // Find key info
    for (const key of ['Customer', 'Project', 'Products', 'Operation']) {
        const infoM = body.match(new RegExp(`${key}[\\s:]*([^.!?\\n]{15,200})`));
        if (infoM) {
            let val = infoM[1].trim().replace(/^\s*(:|-|—)\s*/, '');
            console.log(`  ${key}: ${val.substring(0, 180)}`);
        }
    }
    
    console.log();
}
